crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -1663,4 +1663,33 @@ mod tests {
 
     #[derive(SystemParam)]
     pub struct UnitParam {}
+
+    #[derive(Resource)]
+    pub struct R<const I: usize>;
```

Comment:
Any reason why you added this over re-using `UnitParam`?

Reply:
I want each field to be a distinct type, to make sure the derive macro doesn't mess up the order of the fields.